scripts/assets/file_content_validation_checks.textproto

```diff
@@ -306,3 +306,50 @@ file_content_checks {
   exempted_file_name: "utility/src/test/java/org/oppia/android/util/math/RealExtensionsTest.kt"
   exempted_file_patterns: "testing/src/main/java/org/oppia/android/testing/junit/.+?\\.kt"
 }
+
+file_content_checks {
+  file_path_regex: "app/src/main/res/values/component_colors.xml|app/src/main/res/values/color_palette.xml"
+  prohibited_content_regex: "name=.+(?<!_color).>@|name=.*[A-Z].*>@"
+  failure_message: "All color declarations in component_color.xml and color_palette.xml should end with _color suffix following snake_case naming convention."
+}
+file_content_checks {
+  file_path_regex: "app/src/main/res/values/color_defs.xml"
+  prohibited_content_regex: "name=.+(?<=color).>#|name=.*[A-Z].*>#"
+  failure_message: "All color declarations in color_defs.xml should be named using snake_case convention and not contain color keyword."
+}
+file_content_checks {
+  file_path_regex: "app/src/main/res/values/component_colors.xml|app/src/main/res/values/color_palette.xml"
+  prohibited_content_regex: ".>#\\w+<\\/color>"
```

Comment:
Should this use ``\\x`` instead?

Reply:
I am not aware about `\\x` use but it does not seem to working as intended when I tried replacing it with `\\w`.

Reply:
Ah, well the potential issue that I see is ``\\x`` checks for hexadecimal vs. ``\\w`` which checks for any letter, I believe. Since color codes can only contain hex characters I think the pattern might be a bit more correct by using ``\\x``.

Reply:
Using \\x actually throws this error.
```
92) testFileContent_componentColors_doesNotHaveHexColorValue_fileContentIsCorrect(org.oppia.android.scripts.regex.RegexPatternValidationCheckTest)
java.util.regex.PatternSyntaxException: Illegal hexadecimal escape sequence near index 5
.>#\x+<\/color>
     ^
        at java.base/java.util.regex.Pattern.error(Pattern.java:2028)
        at java.base/java.util.regex.Pattern.x(Pattern.java:3380)
        at java.base/java.util.regex.Pattern.escape(Pattern.java:2609)
        at java.base/java.util.regex.Pattern.atom(Pattern.java:2286)
        at java.base/java.util.regex.Pattern.sequence(Pattern.java:2210)
        at java.base/java.util.regex.Pattern.expr(Pattern.java:2069)
        at java.base/java.util.regex.Pattern.compile(Pattern.java:1783)
        at java.base/java.util.regex.Pattern.<init>(Pattern.java:1430)
        at java.base/java.util.regex.Pattern.compile(Pattern.java:1069)
        at kotlin.text.Regex.<init>(Regex.kt:89)
        at org.oppia.android.scripts.regex.MatchableFileContentCheck$Companion.createFrom(RegexPatternValidationCheck.kt:244)
        at org.oppia.android.scripts.regex.RegexPatternValidationCheckKt.main(RegexPatternValidationCheck.kt:45)
        at org.oppia.android.scripts.regex.RegexPatternValidationCheckTest.runScript(RegexPatternValidationCheckTest.kt:2164)
        at org.oppia.android.scripts.regex.RegexPatternValidationCheckTest.testFileContent_componentColors_doesNotHaveHexColorValue_fileContentIsCorrect(RegexPatternValidationCheckTest.kt:1877)

FAILURES!!!
Tests run: 92,  Failures: 92
```

Reply:
Ah, this was my bad. ``\\x`` is meant to be used with a specific hex sequence, e.g. ``\\xFF``.

Since we're using Java, extended character classes should be supported. Does ``\\p{XDigit}`` work?